### PR TITLE
Brand Intelligence event is no more miserable (mimic venders), makes mimic code a little better

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Brand Intelligence event is no more miserable (mimic venders)
this means animating venders will return stuff
also makes mimic mob code better

## Why It's Good For The Game
It's super annoying event

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

dead mimics

image

</details>

## Changelog
:cl:
code: made mimic code less bad
tweak: mimic mobs (especially, animating venders) will become back to how it was. For Brand Intelligence event, dead mimics will become a destroyed vending machine
tweak: Brand Intelligence targeted vendors no longer explode when the event happens. These will become a stronger version instead. (i.e. more HP, more damage, etc) Otherwise, vending machine mobs have slightly less HP.
tweak: mimic crate will eat items on where it is spawned - including important items like a nuke disk. Still, nuke disk is trackable, and dead mimic will let things out. (Admins can use this to contain stuff into a mimic crate by spawning that mob - but items should be placed on it.)
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 1]
```diff
@@ -32,23 +32,46 @@
...
+		visible_message("<span class='warning'>[src] ate some stuff!</span>")
+		log_game("Newly created mimic-crate ate item(s): [english_list(ate_something)] in [AREACOORD(src)]")
+		message_admins("Newly created mimic-crate ate item(s): [english_list(ate_something)] in [ADMIN_VERBOSEJMP(src)]")
 	add_overlay("[icon_state]_door")
 
+/mob/living/simple_animal/hostile/mimic/Destroy()
+	var/turf_to_spawn = get_turf(src)
+	// spawns a crate/a closet/a locker, whatever
+	if(spawaning_obj_type && ispath(spawaning_obj_type, /obj/structure/closet))
+		var/obj/structure/closet/crate = new spawaning_obj_type(turf_to_spawn)
```
Comment (itsmeow) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm:67]: what if it's not a closet? nothing is created?

File: code/modules/events/brand_intelligence.dm [hunk 2]
```diff
@@ -58,15 +58,31 @@
...
+			switch(rand(1, 100)) // for 30% chance, they're stronger
+				if(1 to 70) // these are usually weak
+					var/adjusted_health = max(M.health-20, 20) // don't make it negative-health
+					M.health = adjusted_health
+					M.maxHealth = adjusted_health
+				if(71 to 80) // has more health
+					var/bonus_health = 15+rand(1, 7)*5
+					M.health += bonus_health
+					M.maxHealth += bonus_health
+					M.desc += " By the way, this seems to be robust..."
```
Comment (itsmeow) [inline code/modules/events/brand_intelligence.dm]:
```suggestion
					M.desc += " This one seems extra robust..."
```

File: code/modules/events/brand_intelligence.dm [hunk 3]
```diff
@@ -58,15 +58,31 @@
...
+				if(81 to 90) // does stronger damage
+					M.melee_damage += 2+rand(1, 6) // 3~8
+					M.desc += " By the way, this seems to be hurting..."
```
Comment (itsmeow) [inline code/modules/events/brand_intelligence.dm]:
```suggestion
					M.desc += " This one seems extra painful..."
```

File: code/modules/events/brand_intelligence.dm [hunk 4]
```diff
@@ -58,15 +58,31 @@
...
+				if(91 to 100) // moves faster
+					M.move_to_delay /= 2 // just half
+					M.desc += " By the way, this is obviously faster."
```
Comment (itsmeow) [inline code/modules/events/brand_intelligence.dm]:
```suggestion
					M.desc += " This one seems more agile..."
```

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/mimic.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/modules/mob/living/simple_animal/hostile/mimic.dm:67]: nothing will be created. it's specified when a mimic is crate and they should remain a crate as broken after they die

**Review by itsmeow — APPROVED** [review]
Brand Intelligence is one of those events that makes me annoyed every time I see it and yet I need to roleplay like it's some brand new fresh thing, and that makes me hate it. This should make it less awful for everyone involved.

**Pull request merged**